sdk/messaging/azservicebus/internal/amqpLinks.go

```diff
@@ -66,6 +67,12 @@ type AMQPLinks interface {
 
 	// Prefix is the current logging prefix, usable for logging and continuity.
 	Prefix() string
+
+	// Tracer returns the tracer for the AMQPLinks instance.
+	Tracer() tracing.Tracer
+
+	// SetTracer sets the tracer for the AMQPLinks instance.
+	SetTracer(tracing.Tracer)
```

Comment:
Does AMQPLinks needs to own a Tracer object or should it just be passed in as an argument to each function call?

Reply:
For this I am open to your preference :) Currently the tracer starts a span at the `Retry()` function level. So we can either have it in the amqpLink layer, or keep it in the Sender/Receiver layer and passing it as an argument to each function call all the way down to the `Retry()` function level. Which option do you think is more appropriate here?

Reply:
Let me know if it gets too gnarly though.

Reply:
Updated to move the tracers 1 level up. Now they live in `Sender`, `Receiver` and `Namespace`